operator/pkg/name/name.go

```diff
@@ -77,8 +84,32 @@ var (
 	DeprecatedNames = []ComponentName{
 		InjectorComponentName,
 	}
-	allComponentNamesMap        = make(map[ComponentName]bool)
-	deprecatedComponentNamesMap = make(map[ComponentName]bool)
+	AllLegacyAddonComponentNames = []ComponentName{
+		PrometheusComponentName,
+		KialiComponentName,
+		GrafanaComponentName,
+		TracingComponentName,
+		CoreDNSComponentName,
+	}
+	allComponentNamesMap         = make(map[ComponentName]bool)
+	deprecatedComponentNamesMap  = make(map[ComponentName]bool)
```

Comment:
nit: using a struct{} instead of a bool is slightly nicer for a Set because that's is only 2 states (present or not present) instead of 3 (not present, true, false) - what does it mean to be false? It's also (trivially) more perfomant

Reply:
you mean using an empty struct to replace bool? that's actually better in terms of memory(trivial) but here I just try to align with other implementation for set in this package.